### DISCUSSION TITLE
Set schema version (user_version) after each migration

### DISCUSSION THREAD

**Discussion by maxstubbersfield** [simolus3/drift#2436] [Q&A] [answered]
Hi there,

I am using the `onUpgrade` callback on `MigrationStrategy` to migrate the database one version after the other.

From what I understand, the schema version is set after all the migrations have run successfully. For example, upgrading from v1 to v3 would only update the version to v3 after `onUpgrade` completes. 

I have a migration that could possibly throw an exception so I want to ensure that all the migrations up to that point are not re-run the next time the database is opened. 

Is there any issue in upgrading the version manually within the `onUpgrade` method after each migration completes?
I.e. running the code below after each migration completes successfully
`customStatement('PRAGMA user_version = $version;')`

Thanks!

**Comment by simolus3** [comment, marked as answer]
> From what I understand, the schema version is set after all the migrations have run successfully

That's correct.

> Is there any issue in upgrading the version manually within the `onUpgrade` method after each migration completes?

This will work, yes. I've also added a test for this in b5d565e16427e5e9c30475c4329a5c0ad34fbce2 so you can rely on this continuing to work in the future :)

**Reply by maxstubbersfield** [reply]
Fantastic, many thanks for your help! :)
